nbgrader/server_extensions/formgrader/static/js/gradebook_notebook_submissions.js

```diff
@@ -187,6 +187,20 @@ var loadSubmittedNotebooks = function () {
     });
 };
 
+// button to toggle all student names
+var names_hidden = true;
```

Comment:
If all the names have already been displayed using the eye icons, clicking on the button will do nothing the first time.
Maybe instead of a global variable we could check in the function if some names are hidden.

Reply:
I changed the logic so that the button label is updated properly when pressing individual eye icons. The button's logic then depends on its label.